File: LMS.Monolith/lms-frontend/src/types/student.ts
import type { StudentEnrollment } from './enrollment'

export interface Student {
  id: number
  firstName: string
  lastName: string
  email: string
  enrollments?: StudentEnrollment[]
}

export interface CreateStudentDto {
  firstName: string
  lastName: string
  email: string
}

export interface UpdateStudentDto {
  firstName?: string
  lastName?: string
  email?: string
}

File: LMS.Monolith/LMS.Core/Entities/Student.cs
namespace LMS.Core.Entities
{
    // LMS.Core/Entities/Student.cs
    public class Student
    {
        public int Id { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string Email { get; set; }
        public DateTime DateOfBirth { get; set; }
        public DateTime CreatedDate { get; set; }
        public DateTime? UpdatedDate { get; set; }
        public ICollection<Enrollment> Enrollments { get; set; }
    }
}
